backend/app/authentication.py

```diff
@@ -31,6 +31,16 @@ async def get_current_user(token=Depends(get_current_user_token)) -> dict:
             detail="This endpoint requires authorization. You need to be logged in.",
             headers={"WWW-Authenticate": "Bearer"},
         )
+    return await _get_current_user(token)
+
+
+async def get_current_user_optional(token=Depends(get_current_user_token)) -> dict:
```

Comment:
I don't think this is the best approach. We can keep it for now, but please add a TODO to refactor this part